src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs

```diff
@@ -59,9 +57,7 @@ async Task<ImmutableArray<string>> GetApplyUpdateCapabilitiesCoreAsync()
                 {
                     // We'll query the browser and ask it send capabilities. If the browser does not respond in a short duration, we'll assume something is amiss and return
                     // baseline capabilities.
-                    var response = await context.BrowserRefreshServer.ReceiveAsync(buffer, cancellationToken)
-                        .AsTask()
-                        .WaitAsync(TimeSpan.FromSeconds(15), cancellationToken);
+                    var response = await context.BrowserRefreshServer.ReceiveAsync(buffer, cancellationToken);
```

Comment:
How safe is it to remove this timeout? It was there for a reason?
Looking at the above comment as well as the code, it looks like there is a possibility for this call to hang for a long period of time. Feels like this can stall the execution.

Reply:
I don't think there is a reason for having the timeout at all. The user can cancel any of these calls.

Reply:
If this call times out the rest of dotnet watch functionality is broken.

Reply:
If I run `dotnet watch` on a Bazor app and it is not able to connect to the browser in a given time then is it preferable to terminate `dotnet watch` with a message that it's not able to connect, or wait for the browser indefinitely with "Waiting to connect ..." message printed out to the console?

Reply:
The user can hit Cltr+C any time btw to cancel.

Reply:
I think it is still beneficial to have some timeout and print "unable to connect to browser after waiting for X minutes". But it's your call, @tmat.

Reply:
As long as it's clear to the user that the problem is "we haven't yet managed to connect" then TBH I don't think it matters whether we eventually give up or wait for a ctrl+c.

In both cases it's not working and the user knows what the problem is, so it's more or less equivalent.

Reply:
I have added code that prints "Waiting for application connection ..." warning every 5s until the application connection is established (prints nothing if it takes less than 5s). I also added regular output "Application ready for Hot Reload" once initialization is complete. Although changes can be made to source files earlier they can't be applied until we finish initialization, so having output indicate that seems useful.

![image](https://user-images.githubusercontent.com/41759/190031717-d1d1af54-9664-4844-9395-fe54f29d6efa.png)